### ISSUE TITLE
Incompatible dependency: glob requires Node 20+

### ISSUE THREAD

**Issue by bmish** [igorshubovych/markdownlint-cli#508]
markdownlint-cli depends on glob.

glob v11 requires Node 20+: https://github.com/isaacs/node-glob/blob/main/changelog.md#110

But markdownlint-cli currently specifies that it supports Node >= 18:

https://github.com/igorshubovych/markdownlint-cli/blob/0d9fcb51a54f3b750b911c054b4bd1a590f1b592/package.json#L10C16-L10C18

So the solution would be to downgrade glob until requiring Node 20+.

**Comment by DavidAnson** [comment]
True, but that was 6 months ago and this doesn't seem to have been a problem for CI or users. (I wish Dependabot wouldn't offer updates like this that are semi-incompatible.)

**Comment by DavidAnson** [comment]
@bmish, do you know of a tool that validates this (and could run in CI) or did you just happen to spot this manually?

**Comment by bmish** [comment]
Here's a fix that would catch it:

* https://github.com/igorshubovych/markdownlint-cli/pull/509 

I will note that this isn't really an urgent problem, so I don't mind if you want to:
* Downgrade glob
* Or just wait until [2025-04-30](https://github.com/nodejs/Release#release-schedule) to drop support for Node 18 and merge the fix so it doesn't happen again

**Comment by DavidAnson** [comment]
My current thinking is to wait, but leave the issue open for visibility. :)

**Comment by DavidAnson** [comment]
@bmish, FYI that markdownlint library and CLI2 now both have engine-strict enabled (without issue on Node 18) in their "next" branches.

**Comment by lsh-0** [comment]
> My current thinking is to wait, but leave the issue open for visibility. :)

If we could get a 0.44.0 with a downgraded `glob` that would be helpful, thank you

**Comment by DavidAnson** [comment]
@lsh-0, versions `0.41.0` and earlier of `markdownlint-cli` use the older version of `glob` that declares support for Node `18`. Node `20` and later are supported by `glob` with the latest version of `markdownlint-cli`. (And there have been no reported issues due to this in 6 months.) Is there a compelling case to downgrade?

**Comment by lsh-0** [comment]
> Is there a compelling case to downgrade?

The set of dependencies described by `markdownlint-lint` are incompatible. It doesn't support node18 any more because of `glob@11.x`.

Downgrading `glob` seems the most natural fix but I guess you could also change this line and remove support for node18:

https://github.com/igorshubovych/markdownlint-cli/blob/e9435ddcf9844e850e2c117deaeafd0ca13dae33/package.json#L11

**Comment by DavidAnson** [comment]
I'm not familiar with `markdownlint-lint`, I don't see it on npm? But because there are two different ways to address this issue today (see my comment above) and no known problems with the current (inconsistent) configuration, how necessary is it to downgrade `glob` within this project?

**Comment by lsh-0** [comment]
> I'm not familiar with markdownlint-lint, I don't see it on npm

Typo, I meant `markdownlint-cli`, this project.

> and no known problems with the current (inconsistent) configuration, how necessary is it to downgrade glob within this project?

There are problems. We are having problems. There are two solutions or you can do nothing.

**Comment by DavidAnson** [comment]
Please give an example of a problem that is happening because of this situation along with the steps to reproduce it.

**Comment by bmish** [comment]
I would suggest just using an older version of `markdownlint-cli` until the problem is solved.
